File: java_projects/catgame/src/com/quetzal/catgame/Draw.java
package com.quetzal.catgame;

public class Draw {
    private final char TOP_LEFT;
    private final char TOP_RIGHT;
    private final char BOTTOM_LEFT;
    private final char BOTTOM_RIGHT;
    private final char HORIZONTAL;
    private final char VERTICAL;
    private final char CROSS;

    public Draw() {
        TOP_LEFT = '\u250C';        // ┌
        TOP_RIGHT = '\u2510';       // ┐
        BOTTOM_LEFT = '\u2514';     // └
        BOTTOM_RIGHT = '\u2518';    // ┘
        HORIZONTAL = '\u2500';      // ─
        VERTICAL = '\u2502';        // │
        CROSS = '\u253C';           // ┼
    }

    private char[][] drawBoard(char[][] board) {
        int rows = 5;
        int columns = 11;

        char[][] result = new char[rows][columns];

        for(int row = 0; row < rows; row++) {
            int position = 0;

            for(int column = 0; column < columns; column++) {
                if(row % 2 == 0) {
                    if(column % 2 == 0) {
                        result[row][column] = ' ';
                    } else {
                        if(column == 3 || column == 7) {
                            result[row][column] = VERTICAL;
                        } else {
                            result[row][column] = board[row / 2][position];
                            position++;
                        }
                    }
                } else {
                    if(column == 3 || column == 7) {
                        result[row][column] = CROSS;
                    } else {
                        result[row][column] = HORIZONTAL;
                    }
                }
            }
        }
        return result;
    }

    public void drawSquare(char[][] board) {
        char[][] drawBoard = drawBoard(board);

        int rows = (drawBoard.length) + 2;
        int columns = (drawBoard[0].length) + 2;

        for(int height = 0; height <= rows; height++) {
            for(int width = 0; width <= columns; width++) {
                if(height == 0) {
                    if(width % 2 == 0) {
                        System.out.print(" ");
                    } else {
                        if(width == 3 || width == 7 || width == 11) {
                            System.out.print(width / 4);
                        } else {
                            System.out.print(' ');
                        }
                    }
                } else if(height == 1) {
                    if(width == 0) {
                        System.out.print(' ');
                    } else if(width == 1) {
                        System.out.print(TOP_LEFT);
                    } else if(width == columns) {
                        System.out.print(TOP_RIGHT);
                    } else {
                        System.out.print(HORIZONTAL);
                    }
                } else if(height >= rows) {
                    if(width == 0) {
                        System.out.print(' ');
                    } else if(width == 1) {
                        System.out.print(BOTTOM_LEFT);
                    } else if(width == columns) {
                        System.out.print(BOTTOM_RIGHT);
                    } else {
                        System.out.print(HORIZONTAL);
                    }
                } else {
                    if(width == 0) {
                        if(height % 2 == 0) {
                            System.out.print(height / 2 - 1);
                        } else {
                            System.out.print(' ');
                        }
                    } else if(width == 1) {
                        System.out.print(VERTICAL);
                    } else if(width == columns) {
                        System.out.print(VERTICAL);
                    } else {
                        System.out.print(drawBoard[height - 2][width - 2]);
                    }
                }
            }
            System.out.println();
        }
    }
}

File: python_projects/catgame/draw.py
class Draw:
    def __init__(self):
        self.TOP_LEFT = '\u250C'        # ┌
        self.TOP_RIGHT = '\u2510'       # ┐
        self.BOTTOM_LEFT = '\u2514'     # └
        self.BOTTOM_RIGHT = '\u2518'    # ┘
        self.HORIZONTAL = '\u2500'      # ─
        self.VERTICAL = '\u2502'        # │
        self.CROSS = '\u253C'           # ┼


    def draw_board(self, board):
        rows = 5
        columns = 11

        result = []

        for row in range(rows):
            row_elements = []
            position = 0

            for col in range(columns):
                if row % 2 == 0:
                    if col % 2 == 0:
                        row_elements.append(' ')
                    else:
                        if col == 3 or col == 7:
                            row_elements.append(self.VERTICAL)
                        else:
                            row_elements.append(board[row // 2][position])
                            position += 1
                else:
                    if col == 3 or col == 7:
                        row_elements.append(self.CROSS)
                    else:
                        row_elements.append(self.HORIZONTAL)
            result.append(row_elements)
        return result


    def draw_square(self, board):
        draw_board = self.draw_board(board)

        rows = len(draw_board) + 3
        columns = len(draw_board[0]) + 3

        for height in range(rows):
            for width in range(columns):
                if height == 0:
                    if width % 2 == 0:
                        print(' ', end='')
                    else:
                        if width == 3 or width == 7 or width == 11:
                            print(width // 4, end='')   # Imprime la coordenada
                        else:
                            print(' ', end='')
                elif height == 1:
                    # Dibuja la barra superior del cuadrado
                    if width == 0:
                        print(' ', end='')
                    elif width == 1:
                        print(self.TOP_LEFT, end='')
                    elif width >= (columns - 1):
                        print(self.TOP_RIGHT, end='')
                    else:
                        print(self.HORIZONTAL, end='')
                elif height == rows - 1:
                    # Dibuja la barra inferior del cuadrado
                    if width == 0:
                        print(' ', end='')
                    elif width == 1:
                        print(self.BOTTOM_LEFT, end='')
                    elif width >= (columns - 1):
                        print(self.BOTTOM_RIGHT, end='')
                    else:
                        print(self.HORIZONTAL, end='')
                else:
                    if width == 0:
                        if height % 2 == 0:
                            print(height // 2 - 1, end='')     # Imprime la coordenada
                        else:
                            print(' ', end='')
                    elif width == 1:
                        print(self.VERTICAL, end='')
                    elif width >= (columns - 1):
                        print(self.VERTICAL, end='')
                    else:
                        # Dibuja el tablero en el interior del cuadro
                        print(draw_board[height - 2][width - 2], end='')
            print()
